src/md.ml

```diff
@@ -1,14 +1,40 @@
 let langs = [ "ocaml"; "dune"; "opam"; "sh"; "shell"; "bash" ]
 
-let transform blocks =
-  let f (b : 'a Omd.block) =
+let rec transform_block block =
+  let f (b : Cmarkit.Block.t) =
     match b with
-    | Omd.Code_block (attr, lang, src) -> (
-        if not (List.mem lang langs) then b
-        else
-          match Syntax.src_code_to_html ~lang ~src with
-          | Ok html -> Omd.Html_block (attr, html)
-          | Error (`Msg s) -> failwith s)
+    | Cmarkit.Block.Code_block (node, meta) -> (
+        let info =
+          Cmarkit.Block.Code_block.info_string node |> Option.map fst
+        in
+        let code =
+          Cmarkit.Block.Code_block.code node
+          |> List.map Cmarkit.Block_line.to_string
+          |> String.concat "\n"
+        in
+        match
+          Option.bind info Cmarkit.Block.Code_block.language_of_info_string
+        with
+        | Some (lang, _) -> (
+            if not (List.mem lang langs) then b
+            else
+              match Syntax.src_code_to_html ~lang ~src:code with
+              | Ok html ->
+                  (* let buf = Buffer.create 128 in
+                     Cmarkit_html.buffer_add_html_escaped_string buf html; *)
+                  let h = Cmarkit.Block_line.list_of_string html in
+                  Cmarkit.Block.Html_block (h, meta)
+              | Error (`Msg s) -> failwith s)
+        | _ -> b)
+    | Cmarkit.Block.Blocks (bs, meta) ->
+        Cmarkit.Block.Blocks (List.map transform_block bs, meta)
     | _ -> b
```

Comment:
There's a good chance this is missing some items to recurse into here

Reply:
Fixed in [21a9178](https://github.com/patricoferris/hilite/pull/11/commits/21a9178ab9f12e1f83f1292b69d8980f2d9adcfd) by using Cmarkit's built-in mapper